cmd/skywire-cli/commands/config/gen.go

```diff
@@ -98,10 +201,158 @@ func init() {
+	if err == nil {
+		return strings.TrimSpace(z)
+	}
+	return ""
+}
+
+func scriptExecBool(s string) bool {
+	if visorconfig.OS == "windows" {
+		var variable string
+		if strings.Contains(s, ":-") {
+			parts := strings.SplitN(s, ":-", 2)
+			variable = parts[0] + "}"
+		} else {
+			variable = s
+		}
+		out, err := script.Exec(fmt.Sprintf(`powershell -c '$SKYENV = "%s"; if ($SKYENV -ne "" -and (Test-Path $SKYENV)) { . $SKYENV }; echo %s"`, skyenvfile, variable)).String()
+		if err == nil {
+			if (out == "") || (out == variable) {
+				return false
+			}
+			b, err := strconv.ParseBool(strings.TrimSpace(strings.TrimRight(out, "\n")))
+			if err == nil {
+				return b
+			}
+		}
+		return false
+	}
+	z, err := script.Exec(fmt.Sprintf(`bash -c 'SKYENV=%s ; if [[ $SKYENV != "" ]] && [[ -f $SKYENV ]] ; then source $SKYENV ; fi ; printf "%s"'`, skyenvfile, s)).String()
+	if err == nil {
+		b, err := strconv.ParseBool(z)
+		if err == nil {
+			return b
+		}
+	}
+
+	return false
+}
+
+func scriptExecArray(s string) string {
+	if visorconfig.OS == "windows" {
+		variable := s
+		if strings.Contains(variable, "[@]}") {
+			variable = strings.TrimRight(variable, "[@]}")
+			variable = strings.TrimRight(variable, "{")
+		}
+		out, err := script.Exec(fmt.Sprintf(`powershell -c '$SKYENV = "%s"; if ($SKYENV -ne "" -and (Test-Path $SKYENV)) { . $SKYENV }; foreach ($item in %s) { Write-Host $item }'`, skyenvfile, variable)).Slice()
+		if err == nil {
+			if len(out) != 0 {
+				return ""
+			}
+			return strings.Join(out, ",")
+		}
+	}
+	y, err := script.Exec(fmt.Sprintf(`bash -c 'SKYENV=%s ; if [[ $SKYENV != "" ]] && [[ -f $SKYENV ]] ; then source $SKYENV ; fi ; for _i in %s ; do echo "$_i" ; done'`, skyenvfile, s)).Slice()
+	if err == nil {
+		return strings.Join(y, ",")
+	}
+	return ""
+}
+
+func scriptExecInt(s string) int {
+	if visorconfig.OS == "windows" {
+		var variable string
+		if strings.Contains(s, ":-") {
+			parts := strings.SplitN(s, ":-", 2)
+			variable = parts[0] + "}"
+		} else {
+			variable = s
+		}
+		out, err := script.Exec(fmt.Sprintf(`powershell -c '$SKYENV = "%s"; if ($SKYENV -ne "" -and (Test-Path $SKYENV)) { . $SKYENV }; echo %s"`, skyenvfile, variable)).String()
+		if err == nil {
+			if (out == "") || (out == variable) {
+				return 0
+			}
+			i, err := strconv.Atoi(strings.TrimSpace(strings.TrimRight(out, "\n")))
+			if err == nil {
+				return i
+			}
+			return 0
+		}
+		return 0
+	}
+	z, err := script.Exec(fmt.Sprintf(`bash -c 'SKYENV=%s ; if [[ $SKYENV != "" ]] && [[ -f $SKYENV ]] ; then source $SKYENV ; fi ; printf "%s"'`, skyenvfile, s)).String()
+	if err == nil {
+		if z == "" {
+			return 0
+		}
+		i, err := strconv.Atoi(z)
+		if err == nil {
+			return i
+		}
+	}
+	return 0
+}
+
 var genConfigCmd = &cobra.Command{
 	Use:   "gen",
 	Short: "Generate a config file",
+	Long: func() string {
+		if visorconfig.OS == "linux" {
```

Comment:
I dont quite understand the purpose of this. The code is hard to read because of the interspersed multiline strings. But it looks to me like we are returning info here whether or not we can detect the Skyenv file set via env. But we only do this if the user is running it on Linux.